db/models.js

```diff
@@ -1,27 +1,64 @@
-const  db  = require ("./connection")
+const db = require("./connection")
 
 
 const fetchCategories = () => {
     let sqlString = `SELECT * FROM categories`
     return db.query(sqlString)
-    .then (({rows}) => rows)
+        .then(({ rows }) => rows)
 }
 
 
 const fetchReviews = () => {
-    
-    let sqlString = `SELECT reviews.*, count(*) AS comment_count 
+
+    let sqlString = `SELECT reviews.*, count(comments.review_id) AS comment_count 
         FROM reviews 
             FULL JOIN comments ON reviews.review_id = comments.review_id
         GROUP BY reviews.review_id
         ORDER BY reviews.created_at desc;`
-        
-        return db.query(sqlString)
-        .then (({rows}) => rows)
-    }
+
+    return db.query(sqlString)
+        .then(({ rows }) => rows)
+}
+
+const fetchReviewByID = (review_id) => {
+    if (isNaN(review_id)) {return Promise.reject({status : 400, msg : "Not Found"})}
```

Comment:
This is sufficient for how we want our server to behave but it is worth pointing out that postgres would throw an error here regardless as the review_id has to be an INT. We could use that throw error and pass it on to some error handling middleware.